tests/core/builder-tools/test_chain_initializer.py

```diff
@@ -0,0 +1,118 @@
+import time
+
+import pytest
+
+from cytoolz import pipe
+
+from eth import constants
+from eth.chains.base import Chain
+from eth.tools.builder.chain import (
+    frontier_at,
+    genesis,
+)
+
+
+@pytest.fixture
+def chain_class():
+    return frontier_at(0)(Chain)
+
+
+def test_chain_builder_initialize_chain_default(chain_class):
+    chain = pipe(
+        chain_class,
+        genesis(),
+    )
+
+    header = chain.get_canonical_head()
+    assert header == chain.get_canonical_block_by_number(0).header
+
+    assert header.parent_hash == constants.GENESIS_PARENT_HASH
+    assert header.uncles_hash == constants.EMPTY_UNCLE_HASH
+    assert header.coinbase == constants.GENESIS_COINBASE
+    assert header.state_root == constants.BLANK_ROOT_HASH
+    assert header.transaction_root == constants.BLANK_ROOT_HASH
+    assert header.receipt_root == constants.BLANK_ROOT_HASH
+    assert header.bloom == 0
+    assert header.difficulty == 1
+    assert header.block_number == constants.GENESIS_BLOCK_NUMBER
+    assert header.gas_limit == constants.GENESIS_GAS_LIMIT
+    assert header.gas_used == 0
+    # account for runtime.  should run in less than a second and should be
+    # effectively "now"
+    assert abs(header.timestamp - time.time()) < 1
+    assert header.extra_data == constants.GENESIS_EXTRA_DATA
+    assert header.mix_hash == constants.GENESIS_MIX_HASH
+    assert header.nonce == constants.GENESIS_NONCE
+
+
+ADDRESS_A = b'a' + b'\0' * 19
+ADDRESS_B = b'b' + b'\0' * 19
+
+
+def test_chain_builder_initialize_chain_with_state_simple(chain_class):
+    chain = pipe(
+        chain_class,
+        genesis(
+            state=((ADDRESS_A, 'balance', 1),),
```

Comment:
Maybe this would be cleaner as a tuple of objects representing accounts?

Something like:
```py
   chain = pipe(
        chain_class,
        genesis(
            state=(
                  Account(address=ADDRESS_A, balance=1),
            ),
        )
    )

class Account(typing.NamedTuple):
  address: eth_typing.Address
  balance: int = 0
  code: bytes = b''
  ...
```

_(TIL that `NamedTuple` [supports default values](https://docs.python.org/3/library/typing.html#typing.NamedTuple))_

Reply:
ooooooh! :smile_cat: :smile_cat: :smile_cat: :smile_cat: :smile_cat: :smile_cat: :smile_cat: :smile_cat: :smile_cat: :smile_cat: :smile_cat: :smile_cat:

Reply:
Going to save this for a subsequent PR.  That type of named tuple is only supported in python3.6 and this code has to support 3.5.